python.d/neo.chart.py

```diff
@@ -0,0 +1,97 @@
+#     code your query to return two results per line (label , value)
+
+
+# Warning
+#    charts take time to appear in the browser, also dont forget to clean the cache
+#    see /var/log/netdata/error.log to get some info and know if your code is running
+#    (python.d exited with -1  is bad news)
+
+
+from bases.FrameworkServices.SimpleService import SimpleService
+from neo4j.v1 import GraphDatabase
+
+
+ORDER = ['monnodes','monlabels','monrels']
+
+CHARTS = {
+    'monnodes': {
+        'options': [None, 'Nodes count', 'Nodes', 'Node count', 'neo4j', 'line'],
+        'lines': [
+             ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    },
+    'monrels': {
+        'options': [None, 'Relations count', 'Nodes', 'Most present relations', 'neo4j', 'line'],
+        'lines': [
+             ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    },
+    'monlabels': {
+        'options': [None, 'Relations count', 'Nodes', 'Most present labels', 'neo4j', 'line'],
+        'lines': [
+            ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    }
+
+
+}
+
+update_every = 10
+priority = 99000
+retries = 10
+
+
+class Service(SimpleService):
+
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        host = self.configuration.get('host')
+        boltport = self.configuration.get('boltport','7687')
+        uri = "bolt://" + host + ":" + boltport
+        self.neodriver = GraphDatabase.driver(uri, auth=(self.configuration.get('user'), self.configuration.get('pwd') ))
+        
+
+    @staticmethod
+    def check():
+        return True
+
+    def get_data_from_cypher_query(self, chartName,data, neosession, cypherQuery):
+        try:
+            with neosession.begin_transaction() as tx:
+                for neorecord in tx.run(cypherQuery):
+                    if neorecord["label"] not in self.charts[chartName]:
+                        self.charts[chartName].add_dimension([ neorecord["label"] ])
+                    data[ neorecord["label"] ]  = neorecord["cnt"] 
+            tx.close()
+        except Exception:
+            return None
+   
+
+
+    def get_data(self):
+        data = dict()
+        neosession = self.neodriver.session()
+
+        # nodes (total)
+        self.get_data_from_cypher_query( 'monnodes',data, neosession, "MATCH (n) RETURN 'Total' AS label, count(n) AS cnt")
+
+        # relations (top 5)
+        self.get_data_from_cypher_query( 'monrels',data, neosession, "MATCH (n)-[r]-(m) RETURN type(r) as label, count(r) AS cnt ORDER BY cnt DESC LIMIT 5")
+        
+        # labels (top 5)
+        self.get_data_from_cypher_query( 'monlabels',data, neosession, "MATCH (a) WITH DISTINCT LABELS(a) AS temp, COUNT(a) AS tempCnt UNWIND temp AS label RETURN label, SUM(tempCnt) AS cnt ORDER BY cnt DESC LIMIT 5")
+       
+        neosession.close()
```

Comment:
How expensive is it to re-create a session each time you want to fetch data?  Is this a requirement of the protocol itself, or is it possible to reuse the session?